### ISSUE TITLE
Why does df.merge(x) not work when x is a pd.Series? 

### ISSUE THREAD

**Issue by zfrenchee** [pandas-dev/pandas#23250]
#### Code Sample, a copy-pastable example if possible

```
s = pd.Series([10, 9, 11], index=['a', 'b', 'c'], name='w')

df = pd.DataFrame([[3, 4, 5], [6, 7, 8], ['b', 'a', 'c']], index=['x', 'y', 'z'], columns=['p', 'q', 'r']).transpose()

df.merge(s, how='left', left_on='z', right_index=True)  # fails

df.merge(s.to_frame(), how='left', left_on='z', right_index=True)  # succeeds

```

We can all keep using .to_frame(), but I feel like the desired behavior of merging a dataframe with a series is well defined. 

Food for thought...

**Comment by mroeschke** [comment]
This feature will be available in the next release (v0.24.0) or currently on master. xref https://github.com/pandas-dev/pandas/pull/21223

```
In [9]: pd.__version__
Out[9]: '0.24.0.dev0+780.g145c2275e'

In [10]: s
Out[10]:
a    10
b     9
c    11
Name: w, dtype: int64

In [11]: df
Out[11]:
   x  y  z
p  3  6  b
q  4  7  a
r  5  8  c

In [12]: df.merge(s, how='left', left_on='z', right_index=True)
Out[12]:
   x  y  z   w
p  3  6  b   9
q  4  7  a  10
r  5  8  c  11
```

**Comment by zfrenchee** [comment]
@mroeschke incredible!
